### DISCUSSION TITLE
GLTF with repeated material not receiving shadows

### DISCUSSION THREAD

**Discussion by cpheinrich** [google/filament#6535] [General]
We recently added a repeated floor material to our room mode scans, but it seems like now the floor is not receiving shadows (see the shadow from the desk does not cast on the floor now for the one with the material).

There is no other difference in the viewer -- same lights and shadow configs. Is there something I need to be doing differently to have an object with a material receive shadows? 

![IMG_4557 2](https://user-images.githubusercontent.com/12475454/217114620-c1064210-879c-43a8-9c2b-6d01da8d8fba.PNG)
![IMG_4558 2](https://user-images.githubusercontent.com/12475454/217114624-89b265db-c288-4efb-b4a0-4f6906b84109.PNG)

Here is the .glb file:

[office_no_shadow.glb.zip](https://github.com/google/filament/files/10670176/office_no_shadow.glb.zip)

**Comment by romainguy** [comment]
What's your shadows configuration? Sunlight shadows work fine:
<img width="1136" alt="Screenshot 2023-02-06 at 4 39 30 PM" src="https://user-images.githubusercontent.com/869684/217118893-0b1d2d11-9265-4542-903e-1403e9f49f1d.png">

**Comment by cpheinrich** [comment]
Thanks for the reply! We're using VSM shadows with these options, anything look amiss there?

<img width="792" alt="Screen Shot 2023-02-06 at 5 37 35 PM" src="https://user-images.githubusercontent.com/12475454/217125909-0e1bcfe6-2803-4280-aacc-e3e5bbf298c1.png">

**Reply by romainguy** [reply]
It looks fine but I'm not even sure if the shadow is missing from your screenshot. Can you try to reduce/disable the blur?

**Comment by cpheinrich** [comment]
![IMG_845252E418F1-1](https://user-images.githubusercontent.com/12475454/217152884-ed5cd3d5-8a98-4fc4-b250-330fd4b3974f.jpeg)
I turned off the blur and cranked up the directional light. You can see the hard shadows on the walls that stop on the floor. If it is working in the default filament viewer then we must have something wrong on our end somewhere -- that's a good clue and should help us track it down. Thanks!

**Reply by romainguy** [reply]
That's strange, but it could be an issue with the transpilation of shaders to Metal. @bejado have you seen this before?

**Reply by bejado** [reply]
I haven't seen something like this before. @cpheinrich if you're able to upload a Metal frame capture I'd be happy to take a look.

**Reply by cpheinrich** [reply]
I tested on other platforms, and it still reproducing there, so likely not an issue with metal only. We're going to take a closer look at other areas of our code to see if anything else might be effecting this. Thanks for the help!
